plugins/panels/similarity_search/__init__.py

```
@@ -338,6 +345,18 @@ def _get_brain_keys(self, ctx):
             brain_keys = []
 
+        # Prune brain keys whose runs have no persisted results.
+        if brain_keys:
+            try:
+                run_docs = BrainMethod._get_run_docs(dataset)
+                brain_keys = [
+                    key
+                    for key in brain_keys
+                    if key in run_docs and run_docs[key].results
+                ]
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)